guacml/pipeline.py

```diff
@@ -0,0 +1,119 @@
+import base64
+
+from guacml.dataset import Dataset
+from guacml.preprocessing.column_analyzer import ColumnAnalyzer
+from guacml.step_tree.step_tree import StepTree
+from guacml.util import deep_copy, get_fully_qualified_class_name, get_class_from_string
+
+
+class Pipeline:
+
+    @staticmethod
+    def from_tree(guac, model_name):
+        mr = guac.model_results[model_name]
+        tree = guac.tree.get_subtree_upto(model_name)
+        config = tree.config
+        config['column_types'] = mr.metadata.type.to_dict()
+        logger = guac.logger
+        model = mr.model.copy(config)
+        features = list(mr.features)
+
+        return Pipeline(model_name, tree, config, logger, model, features)
+
+    def __init__(self, name, tree, config, logger, model, features):
+        self.name = name
+        self.tree = tree
+        self.config = config
+        self.logger = logger
+        self.model = model
+        self.features = features
+
+    def transform(self, x):
+        step_name = self.tree.root_name
+        metadata = ColumnAnalyzer(self.config, self.logger).analyze(x)
```

Comment:
That'd be really nice to have, but I'm a bit afraid of touching column analyzer right now, because it's pretty convoluted. But I definitely like the idea of having transformers, which are collected during `run()`, and then we can replay them on a different dataset.